api/server.go

```diff
@@ -0,0 +1,90 @@
+// Package server provides a server object that represents the GoCert backend
+package server
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"gopkg.in/yaml.v3"
+)
+
+type ConfigYAML struct {
+	KeyPath  string
+	CertPath string
+	DBPath   string
+	Port     int
+}
+
+type Config struct {
+	Key    []byte
+	Cert   []byte
+	DBPath string
+	Port   int
+}
+
+func ValidateConfigFile(filePath string) (Config, error) {
+	validationErr := errors.New("config file validation failed: ")
+	config := Config{}
+	configYaml, err := os.ReadFile(filePath)
+	if err != nil {
+		return config, errors.Join(validationErr, err)
+	}
+	c := ConfigYAML{}
+	if err := yaml.Unmarshal(configYaml, &c); err != nil {
+		return config, errors.Join(validationErr, err)
+	}
+	cert, err := os.ReadFile(c.CertPath)
+	if err != nil {
+		return config, errors.Join(validationErr, err)
+	}
+	key, err := os.ReadFile(c.KeyPath)
+	if err != nil {
+		return config, errors.Join(validationErr, err)
+	}
+	if _, err := os.OpenFile(c.DBPath, os.O_CREATE|os.O_RDONLY, 0644); err != nil {
+		return config, errors.Join(validationErr, err)
+	}
+	config.Cert = cert
+	config.Key = key
+	config.DBPath = c.DBPath
+	config.Port = c.Port
+	return config, nil
+}
+
+// NewServer creates a new http server with handlers that Go can start listening to
+func NewServer(certificate, key []byte, port int) (*http.Server, error) {
+	serverCerts, err := tls.X509KeyPair(certificate, key)
+	if err != nil {
+		return nil, err
+	}
+	router := http.NewServeMux()
+	router.HandleFunc("GET /", HelloWorld)
+
+	v1 := http.NewServeMux()
+	v1.Handle("/v1/", http.StripPrefix("/v1", router))
+
+	s := &http.Server{
```

Comment:
Please avoid 1 letter variables

Reply:
1 letter variable names are actually [encouraged in some cases in Go](https://google.github.io/styleguide/go/decisions#single-letter-variable-names). `s` is actually the example variable name in Go's own docs [here](https://github.com/golang/go/blob/c71d2a8d8fcb50ed8ab0e1c4af211a0dcea16a75/src/net/http/doc.go#L76).

I can still change it if we don't prefer it.